allennlp/modules/drop_connect.py

```diff
@@ -83,8 +97,15 @@ def forward(self, *args):
             match.module._parameters[match.parameter_name] = F.dropout(
                 raw_parameter, p=self._dropout, training=self.training
             )
-        # Call the top-level module on the inputs.
-        output = self._module(*args)
+        # Avoid the warning of
+        # `/pytorch/aten/src/ATen/native/cudnn/RNN.cpp`:1238:
+        # `UserWarning`: RNN module weights are not part of single contiguous chunk of memory.
+        # This means they need to be compacted at every call, possibly greatly increasing memory
+        # usage. To compact weights again call `flatten_parameters()`.
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            # Call the top-level module on the inputs.
+            output = self._module.forward(*args)
```

Comment:
Why use .forward() instead of calling the module? Does hook processing cause an issue.

Reply:
May be not. I haven't really made up my mind on this. On one hand it feels consistent to have `forward()`, but on the other hand it is a composition rather than an inheritance, so it's confusing for me whether leaving all extra steps of the hook for the outermost module or making it happen here.